### PR TITLE
Added day and week timetable

### LINKED ISSUE #48: As a user, I want to be able to view my timetable quickly so that I am aware of my classes and can prepare for them quickly.

### PR DESCRIPTION
**Pull request by Aseanseen**
Awaiting addition of code from Module/ModuleManager.
Do not merge

### CODE REVIEW

File: src/main/java/seedu/duke/data/ModuleManager.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import seedu.duke.exception.ModuleNotProvidedException;
 import seedu.duke.ui.TextUi;
 
+import java.util.ArrayList;
```
Comment (SimJunYou) [inline src/main/java/seedu/duke/data/ModuleManager.java:8]: What are these edits for?

File: src/main/java/seedu/duke/command/timetable/TimeTableCommand.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import seedu.duke.data.Module;
+import seedu.duke.data.ModuleManager;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import static seedu.duke.ui.TextUi.DIVIDER_LINE;
+
+public class TimeTableCommand extends Command {
```
Comment (SimJunYou) [inline src/main/java/seedu/duke/command/timetable/TimeTableCommand.java]: Command classes should only call methods or store and pass information, and should not have standalone methods. Ideally, your timetable command would call 'TimeTable.generateWeekTable()' (and others) and not have its own method for doing so.

File: src/main/java/seedu/duke/command/timetable/TimeTableCommand.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        for (Module modCode : modCodeList) {
+            // Same day as today
+            if (isSameWeek(modCode)) {
+                weekModList.add(modCode);
+            }
+        }
+        return weekModList;
+    }
+
+    private boolean isSameDayAsToday(Module modCode) {
```
Comment (SimJunYou) [inline src/main/java/seedu/duke/command/timetable/TimeTableCommand.java]: Will roll these methods into the new Lesson class

File: src/main/java/seedu/duke/command/timetable/TimeTableCommand.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        return now.toLocalDate().isEqual(modCode.toLocalDate());
+    }
+
+    private boolean isSameWeek(Module modCode) {
+        // todo Module toLocalDate function with a LocalDateTime
+        return (now.isBefore(modCode.toLocalDateTime()) && now.plusDays(6).isAfter(modCode.toLocalDateTime()));
+    }
+
+    public String dayTimeTable() {
+        ArrayList<Module> dayModList = getDayModCodeList();
```
Comment (SimJunYou) [inline src/main/java/seedu/duke/command/timetable/TimeTableCommand.java]: The logic in this method (and the next method below) will be moved to the new TimeTable class

**Review by SimJunYou — CHANGES_REQUESTED** [review]
Standby for creation of Lesson and TimeTable classes

File: src/main/java/seedu/duke/data/ModuleManager.java (hunk 1 above)
Comment (Aseanseen) [reply src/main/java/seedu/duke/data/ModuleManager.java:8]: Unintended edits

File: src/main/java/seedu/duke/command/timetable/TimeTableCommand.java (hunk 2 above)
Comment (Aseanseen) [reply src/main/java/seedu/duke/command/timetable/TimeTableCommand.java]: Ok, will make amends

**Review by SimJunYou — APPROVED** [review]
LGTM

**Comment by Aseanseen** [comment]
There is some bug with this PR, checks are running forever.

**Pull request closed without merging**